### PR TITLE
NRPT-733: import FLNRO inspections from FME csv output

### PR DESCRIPTION
**Pull request by mtCarto**
- [NRPT-733](https://bcmines.atlassian.net/browse/NRPT-733)

To test you will need to have the S3 DEV variables set in your api environment. 
Task should complete succesfully, re-uses existing duplication checks + record logic.

To trigger import task:

POST http://localhost:3000/api/task
body
`
{"taskType":"fmeImport", "dataSourceType":"nris-flnr-csv", "recordTypes": ["Inspection"]}
`

### CODE REVIEW

File: api/openshift/cron.yaml [hunk 1]
```diff
@@ -111,6 +111,12 @@ objects:
                                 -d '{"taskType":"import", "dataSourceType":"nris-epd"}' \
                                 ${NRPTI_API_URL}/task
 
+                      echo -e "Importing NRIS-FLNRO-Inspections\n"
+                              curl -H "Authorization: Bearer ${TOKEN}" \
+                                -H "Content-Type: application/json" \
+                                -d '{"taskType":"csvImport", "dataSourceType":"nris-flnr-csv", "recordTypes": ["Inspection"]}' \
```
Comment (BcGovNeal) [inline api/openshift/cron.yaml:117]: Did you mean taskType = fmeImport?

File: api/src/utils/fme-import-utils.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+});
+const NRIS_FLNRO_KEY = 'fme-exports/NRIS-FLNRO-Inspections.csv';
+
+/**
+ * Stream csv from S3, validate, and parse to json expected by csv-importer
+ * @param {string} dataSourceType which datasource to transform csv to, Eg. nris-flnr-csv
+ * @param {string} recordType record type to target, Eg. Inspection
+ * @returns {string[]}
+ */
+exports.getAndParseFmeCsv = async function(dataSourceType, recordType) {
```
Comment (BcGovNeal) [inline api/src/utils/fme-import-utils.js:24]: Doesn't seem like this needs to be async

File: api/src/utils/csv-helper.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+// Mock has to be setup before csv-helpers is loaded or the S3 connection doesn't use the mock
+awsMock.setSDKInstance(AWS);
+awsMock.mock('S3', 'getObject', Buffer.from(fs.readFileSync(path.resolve(__dirname, '../tests/fixture/mockCsv.csv'))));
+const { readAndParseCsvFile } = require('./csv-helpers');
+
+describe('readAndParseCsvFile', () => {
+  it('returns parsed row if csv is valid ', async () => {
+    const stream = new AWS.S3().getObject('getObject').createReadStream();
+    const result = await readAndParseCsvFile(stream, 'nris-flnr-csv', 'Inspection');
+    expect(result).t
```
Comment (BcGovNeal) [inline api/src/utils/csv-helper.test.js:15]: Leftover code?

**Review by BcGovNeal — APPROVED** [review]

File: api/src/utils/fme-import-utils.js [hunk 4]
```diff
@@ -21,7 +21,7 @@ const NRIS_FLNRO_KEY = 'fme-exports/NRIS-FLNRO-Inspections.csv';
  * @param {string} recordType record type to target, Eg. Inspection
  * @returns {string[]}
  */
-exports.getAndParseFmeCsv = async function(dataSourceType, recordType) {
+exports.getAndParseFmeCsv = function(dataSourceType, recordType) {
```
Comment (marklise) [inline api/src/utils/fme-import-utils.js]: This is an incorrect adjustment to the function.  Inside there is  `readAndParseCsvFile` which is defined as an async function in the utility file.  It is awaited in side the utility, therefore all callers must await as well.

**Review by marklise — CHANGES_REQUESTED** [review]

File: api/src/utils/fme-import-utils.js (hunk 4 above)
Comment (BcGovNeal) [reply api/src/utils/fme-import-utils.js]: Ah yes, I missed that the `readAndParseCsvFile` should have await on it in my review

**Review by marklise — APPROVED** [review]

**Pull request merged**